### PR TITLE
Minimal plugin system

### PR DESCRIPTION
**Pull request by itirabasso**

### CODE REVIEW

File: src/core/config/config-env.ts [hunk 1]
```diff
@@ -55,3 +55,14 @@ export function internalTask<ArgsT extends TaskArguments>(
 }
 
 export const types = argumentTypes;
+
+export function usePlugin(name: string): void {
+  const path = process.cwd() + "/plugins/";
```
Comment (itirabasso) [inline src/core/config/config-env.ts]: I'm not sure about this

Comment (alcuadrado) [reply src/core/config/config-env.ts]: I was about to comment that this restricts the plugin system too much. You can't use npm-installed plugins like this.

`require(name)` should be enough.

File: src/core/config/config-env.ts [hunk 2]
```diff
@@ -55,3 +55,14 @@ export function internalTask<ArgsT extends TaskArguments>(
...
+  require(path + name);
+}
+
+export const extenders: EnvironmentExtender[] = [];
```
Comment (alcuadrado) [inline src/core/config/config-env.ts]: Move this to its own module. Like `src/core/config/tasks-dsl-instance.ts`.

File: src/types.ts [hunk 3]
```diff
@@ -105,3 +105,9 @@ export type GlobalWithBuidlerRuntimeEnvironment = NodeJS.Global & {
 export interface ResolvedFilesMap {
   [globalName: string]: ResolvedFile;
 }
+
+export interface EnvironmentExtension {
```
Comment (alcuadrado) [inline src/types.ts]: This isn't used anywhere

**Pull request merged**